### ISSUE TITLE
Request / Discussion: Select-Object with one property causes many problems for beginners and has few legitimate use cases for anyone. Can that experience be improved for beginners?

### ISSUE THREAD

**Issue by HumanEquivalentUnit** [PowerShell/PowerShell#5237]
I see novices posting variations of this over and over on the internet:

> I have this code: `get-thing | select name` but why are there @{} in my output and how do I get rid of them? Why doesn't the output equal `-eq 'user1'` ? Why doesn't my string split/replace work on it?

And they're too novice to understand the object pipeline well and know what it means to have an object with one property. They want the name, that looks like the way to do it, it looks like it works, but it doesn't actually do what they want.

There's lots of people tripping over this basically-useless edge-case, over and over and over on their way to learning PowerShell well.

Can we come up with a way to smooth that for beginners *before* they know enough to avoid it? (i.e. putting a rule in PSScriptAnalyzer would be too late to help).

It's my suspicion that selecting one thing, `select-object -property OnePropertyHere` , is an anti-pattern, and has so few legitimate use cases that they would have to be contrived to show what it does. 

Ideas:

 - Selecting one thing should throw an exception "you probably don't want to do that, go read a tutorial on the object pipeline". 
 - Selecting one thing should output a warning to the verbose stream.
 - `Select Name` should acted as if you specified `-ExpandProperty Name` and output a value, not an object. This would be a breaking change, but..
   - Then it would be useful for people who *do* know about the object pipeline as well, who still (I speculate) have few real uses for selecting an object with one property. Where the alternatives are currently the laborious `-ExpandProperty`, the convenient but ugly `|% name` or the awkward `(  ).Name`, then simply `select-object -property name` would do what you most likely actually want it to do. (Downside, how would you then specify if you do want an object with one property?)
 - There should be another cmdlet for this use case which is clearly named to show that it won't output an object, because even "select an object and expand a property so the output is not necessarily an object" is a bit awkward when you stare at it for a while. e.g. `select-value` or `select-propertyvalue` or `get-objectproperty` or etc.

**Comment by iSazonov** [comment]
I wonder why a beginner don't read a cmdlet description and examples.

**Comment by markekraus** [comment]
I think changing `select-object` would just move the problem you are trying to address to another cmdlet. The reason why this trips novices up is that the language is object-centric. At some point a novice is going to need to learn that. They just happen to experience it with `select-object` first because i is a common cmdletd.

> It's my suspicion that selecting one thing, select-object -property OnePropertyHere , is an anti-pattern,

It's not, really. You may have an object where only one of the properties is useful. The objects generated by this may be used as seed objects where other properties are added in the pipeline. Or you may be passing an object to a command that takes more than one of your source objects' properties as a `ValueFromPipelineByPropertyName` but only one is correct in the context. or you may simpley need to create a single column CSV with a header to import into an external system (Yammer takes a few single column CSV's, for example)

> Selecting one thing should throw an exception "you probably don't want to do that, go read a tutorial on the object pipeline".

Definitely not for the reason mentioned above.

> Selecting one thing should output a warning to the verbose stream.

I don't think this one is a good idea either.

>`Select Name` should acted as if you specified `-ExpandProperty Name`

That would get confusing with a difference in operation between `Select-Object Porp1` and `Select-Object Prop1, Prop2`.

> There should be another cmdlet for this use case

I don't think it's worth adding a completely other cmdlet for this.

> select an object and expand a property so the output is not necessarily an object

Where do you see that? 

This is probably something that is better fixed with documentation. Perhaps adding an example of this as  common gotcha would be a good start. The `Export-Csv` cmdlet has an example of avoiding the mangling of objects caused by passing objects to `Format-Table` first. i suspect any "fix" done in `Select-Object` would just push that initial exposure to objects on to another cmdlet.

**Comment by RichardSiddaway** [comment]
I can think of a number of reasons for selecting a single property that are extremely valid and don't see any reason to change the current behaviour of select-object

**Comment by HumanEquivalentUnit** [comment]
@RichardSiddaway I cite [your own blog post](https://richardspowershellblog.wordpress.com/2015/12/23/objects-properties-and-values/) where you say "*One thing that seems to keep causing confusion is using Select-Object to pick off one or more properties from a set of objects*", and you blog that it "keeps" causing confusion *9 years* after you started blogging about PowerShell.

You don't see any reason to try and do anything to improve it?

**Comment by RichardSiddaway** [comment]
It's not a problem that technology can fix. Its an education problem. I've seen many people jump into using PowerShell, and other technologies, and make assumptions about how they work or just don't bother to discover how they work. This is where many of the problem you cite come from.

Changing the behaviour of select-object in the way you advocate isn't going to solve those problems as you'll just be moving the problem area to another cmdlet - probably one of more of  sort-object, where-object or foreach-object

**Comment by iSazonov** [comment]
Is our docs well or we need add anything in the cmdlet help to address the Issue?

**Comment by markekraus** [comment]
@iSazonov The docs could be improved with an example to calls out the common gotcha.

**Comment by lzybkr** [comment]
I can see adding a rule to ScriptAnalyzer, but I'm not sure the behavior should be changed.

I will say most places I see this pattern, it is not necessary.

Consistency is important here. For example, if the properties are dynamically selected, selecting one or more properties shouldn't cause weird problems downstream.

**Comment by HumanEquivalentUnit** [comment]
@markekraus "where do you see that?" - imagine you want to write lines of text to a plain text file, and the accepted way was to use `Export-Csv` but tag on the parameter `-ExpandProperty` to tell it not to write a CSV. Is that something you'd expect to be within `Export-Csv`'s remit? Would you be looking there for that feature, or think that was an odd place to put it? With that comparison, would you say `Select-Object` with a parameter is the best place to select non-PSObject output?

**Comment by HumanEquivalentUnit** [comment]
I'll note that the question was really "Can that experience be improved for beginners?" instead of just "change select-object yes/no". The consideration of dynamic (wildcard) property selection is another convincing dealbreaker against that.

@lzy "I will say most places I see this pattern, it is not necessary." - most places you see ExpandProperty is not necessary, or most places you see selecting one property is not necessary?

As another idea for an approach, what if the default output formatter included a line of type information, or other type marker in the output? So when you do `Get-ChildItem c:\path | select-object name` and see a screen full of names, and the header scrolled off, there's still some indication that you have not-text.

StackOverflow has ~400 comments relating to '%expandproperty%' and picking some to see either code suggestions to use ExpandProperty, or people giving thanks for suggesting it, e.g.

```
PostId	Text
42003324	Thanks, included -expandproperty
36187162	`|Select -ExpandProperty HomeDrivePath`
44873829	`Select-Object -ExpandProperty UserName`
36658641	I always forget about `-ExpandProperty`!
45428798	`| select -ExpandProperty SAMAccountName |`
17870097	Thank-you the, -ExpandProperty Node, did it :)
42003324	or select -expandproperty to just get the value
29017197	Thank you for that. `-ExpandProperty`...who knew.
33171965	@DaveSexton `Select  -ExpandProperty Measurements`
39231089	Try `select-object -ExpandProperty DisplayVersion`
38185398	Get-AzureRmLocation | Select -ExpandProperty Location
27942669	Nice. Didn't know about expandProperty. Thanks Duncan.
33017054	Should not it be `Select-Object -ExpandProperty Index`?
33507224	I think it should be select count -ExpandProperty Count.
40345578	`Select-Object mail` ? `Select-Object -ExpandProperty mail`
6375323	You're just emulating a `select -expandproperty size` there.
38338320	try changing that last part to `select -expandproperty ipaddress`
23139592	See my edit and tell me if -ExpandProperty shows the entire value.
25261455	cheers, both options worked. I'll go and read up on -expandproperty
21394652	Alternatively you could do `Select-Object -ExpandProperty identity`.
14139818	Thanks @Kayasax, was spot on !! ( I didn't knew about ExpandProperty)
30718643	Thanks!  `Select-Object -ExpandProperty FullName`  is working for me.
39268354	Use `Select-Object -ExpandProperty` instead of `Select-Object -Property`
18145840	I used -ExpandProperty name to get exactly value named "Administratoren".
29973404	Change `select SAMAccountName` to `select -ExpandProperty SAMAccountName`
43764032	What if you change Select-Object Name to Select-Object -ExpandProperty Name
25395736	Also: `measure-command { ... } | select-object -expandproperty TotalSeconds`
29067832	Have you tried `Select -ExpandProperty Servername` instead of `Format-Table`?
40331593	Hi, try `$listoflogs = Get-EventLog -List | Select-Object -ExpandProperty Log`.
25566906	Do it work with Select-Object -ExpandProperty instead of Select-Object -Property?
33021473	Ah I thought its good for storing as well. Thank you!  -ExpandProperty works good.
17581252	Thanks @jbsmith! The `-ExpandProperty` parameter to `Select-Object` did the trick.
[etc.]
```

and imagining that's the forever-future as one by one, people fall into that trap and have to be helped past it. 

Help.
Use `-expandproperty`
Thanks.

Help.
Use `-expandproperty`
Thanks.

Help.
Use `-expandproperty`
Thanks.

Help.
Use `-expandproperty`
Thanks.

In a world where that is not happening anymore, and Select-Object is not broken, what else could have changed to smooth over that annoyance?

**Comment by markekraus** [comment]
> What else could have changed to smooth over that annoyance?

Forced education camps, maybe? /s

I mean.. this is more about a fundamental difference between PowerShell and other languages and just about every novice (even those experienced in other languages, maybe even especially so) is going to run into this. It's like `return` and "leaky" functions. These are core paradigm differences about the language that can only be learned about either by actually studying the language before coding it, or through trial and error. 

Documentation may help, but only if users read it. A warning would be annoying for a non-novice who knows what they're doing. PSScriptAnalyzer rule would be nice.... if a novice knew about those, they probably already know better.

**Comment by lzybkr** [comment]
@HumanEquivalentUnit - most uses of `Select-Object` actually - with or without `-ExpandProperty`, with one or more property names. But my experience could be highly skewed, so fee free to ignore if you think I'm way off. ;)

**Comment by iSazonov** [comment]
:-) Looking on StackOverflow has ~400 comments "I didn't knew about ExpandProperty" I think we must install PowerShell in "Beginner Mode" with only available Get-Help cmdlet and unlock rest cmdlets only after the beginner pass an exam for 1000 points.

**Comment by markekraus** [comment]
@iSazonov Like a tutorial level in games.

Though, on a more serious note... Maybe if `pwsh` had a `-Tutorial` switch with a guided REPL that would maybe help.

**Comment by iSazonov** [comment]
`powershell /?` already show `PowerShell Online Help https://aka.ms/pscore6-docs`.

**Comment by markekraus** [comment]
@iSazonov Yup. but there is a massive difference between static documentation and an interactive tutorial. There was something similar with learnpython.com and I cen't remember the name but there was one you could run right from the command line. This is more "food for thought" than a serious suggestion.

**Comment by iSazonov** [comment]
I think you should open new Issue for the good suggestion.

**Comment by HumanEquivalentUnit** [comment]
> I mean.. this is more about a fundamental difference between PowerShell and other languages and just about every novice [..] is going to run into this.

Just about everyone, huh? Can some insightful enough change make it a ramp to run up, instead of a wall to run into?

Have you heard the meme that "Vi/Vim is impossible for novices to quit" ? Did that go away because `vimtutor` exists? Not to say documentation and tutors are a bad idea, but what is it that's leading people to think that the text or number they see, is exactly what they have, or to not get to `-ExpandProperty` during the process of typing and using, rather than having to turn to a reference?

With PSReadLine and the changes to conhost , is there any way the default output formatting could change color or style (underline?) to distinguish between when the text you see is representing PSObjects vs strings/value types ?

I don't think this is quite the right problem for documentation - it's one thing to say "I need to login, let me look for a cmdlet option relating to authentication", but if you don't know that you need just one property of an object, you won't go looking for anything like that in the documentation, and if you read past it, you won't recognise the terminology to know that was describing the solution to your problem.

> I think we must install PowerShell in "Beginner Mode" with only available Get-Help cmdlet and unlock rest cmdlets only after the beginner pass an exam for 1000 points.

Are you trying to make it Oracle PowerShell now? 😆 Unlock the PowerShell 2 engine for free, PowerShell 5.1 by entering your $10k certification exam code. Cmdlets, $100/month/10 cmdlets, objects in the pipeline, $10/property, $15/method...

**Comment by RichardSiddaway** [comment]
At what point do you propose we stop trying to make things as simple as possible for new users. Any technology has a learning curve. This is one step on the PowerShell learning curve

**Comment by markekraus** [comment]
@HumanEquivalentUnit I will reiterate: if you make this easier on `Select-Object` you will just be passing the buck on to another cmdlet. This isn't about `Select-Object` but about a fundamental paradigm of PowerShell that needs to be learned one way or another. `Select-Object` is just the first place many users encounter the need to learn this paradigm. This is something that should be addressed with documentation and possibly a tutorial as any technical solution will likely lead to a slippery slope of cascading requests to make changes to the language until this paradigm is either lost or buried and the language no longer PowerShell.

**Comment by HumanEquivalentUnit** [comment]
@markekraus I agree it's a fundamental paradigm change, I disagree about the passing-the-buck or the panic of losing the core of the language. The point is not to hide the paradigm in select-object and pretend it doesn't exist, pushing the discovery elsewhere, so that people ignore it for longer and run into the exact same problem somewhere else. The point of any change would be to reveal the paradigm earlier, to make it so that the paradigm couldn't be ignored that long at all, to always have it visible, to have an interactive lead-in clues to the paradigm and to where to look and what to do about it, helping people learn it rather than not helping. e.g. people who convert to string post messages like `why do I have @{name='file.txt'} and how do I get rid of it` but people who look at the output and it looks like text, *don't even know they have a problem that reading the documentation would help with*.

Like, how this GitHub reply window supports MarkDown formatting, but it has WYSIWYG controls and a preview tab, but Reddit posts support MarkDown formatting, has no indication that it does, and people continually can't format code, and apologise for not knowing how. 

That doesn't mean people no longer have to learn about MarkDown, or that it's too-simple, it means the interface is helpful rather than hidden. Even if you know nothing about it, you can see the buttons are there always on screen as a very big hint that it's not just a plain textbox.

@RichardSiddaway .. at the point where new users aren't suffering in ways that can be ameliorated? At the point where everyone on the planet uses PowerShell so there are no new users? When nobody on the planet has any new suggestions for possible improvements?  When there are no remaining humans with enough interest to discuss any ideas? Why would you assume the time to stop is "roughly now" instead of "in twenty or fifty years"?

**Comment by rkeithhill** [comment]
I wonder if would be useful to have a mechanism that displays tips for interactive users kind of like PSSA supplies tips for scripters.  Well, except that interactive use analysis would be forgiving of using aliases and other wrist-saving techniques.  

Perhaps this is something that could be prototyped as a PSReadline function and bound to a keyboard shortcut.  Press <kbd>Ctrl+i</kbd> to invoke a function that analyzes the current command and provides tips.  That could be interesting.

**Comment by lzybkr** [comment]
Interactive suggestions are easy to extend in the console host - start [here](https://github.com/PowerShell/PowerShell/blob/14af3e12972873676f76b2a456a86c1bf33749b9/src/System.Management.Automation/engine/hostifaces/HostUtilities.cs#L335)

**Comment by markekraus** [comment]
@HumanEquivalentUnit 
>  The point of any change would be to reveal the paradigm earlier, to make it so that the paradigm couldn't be ignored that long at all, 

And you and I are not disagreeing there. But the question is how? Because, changing `select-object` is pretty much a waste of effort, because (as I am repeating myself) it will just move the problem elsewhere not address the root cause. But you mocked our attempts at an interactive solution. So, what is your solution, if any, when a change to `select-object` is off the table?

**Comment by ScriptAutomate** [comment]
@HumanEquivalentUnit 

It sounds like this issue thread is an extension of these:
- Why does `Export-CSV` return `System.Object` in my spreadsheets?
- Why does using `Export-CSV` after `Format-Table` in a pipeline mess all the output up?
- Why doesn't `get-process | select 'CPU(s)'` show anything?
- How come `"Found process: $Process"` outputs `Found process: System.Diagnostics.Process ((sd-pam))` ???
- What's the point of `Get-Member`? Properties and Methods? Wha?
- What is `measure` actually counting?
- What is the pipeline even doing in PowerShell?

In the end, absolutely all of these questions are about **objects and how they are being handled in PowerShell**, and how it is not just text output manipulation. You can't answer any of these without bringing the _object_ into play in an explanation. Even a `String` is an object in PowerShell! As has been explained in this thread, _the issue is in understanding objects_. Look at the cmdlet name of `Select-Object` -- it has `Object` in the name!

You've also rephrased your question for the most part when you said:

> If the type matters and you need to know it - can it be made visible?

`Get-Member` would need to be used. PowerShell gives the best default output it can based on what it believes needs to be presented as value. _"If...you need to know it"_ is a relative/subjective statement in this case. `Select-Object` isn't the _problem_ of what you see on StackOverflow, it is often part of the _solution_ in their use cases by helping them drill down into the values they are looking for in that specific situation. Those specific objects, perhaps, could be debated on how the default formatted output is shown?

Running `help about_Format.ps1xml` will explain a lot about how default formatting to the console is happening as it is, and how it can be customized.

`Get-Process` default console output throws many people off when first starting out with PowerShell. It's not `Select-Object` that needs to be fixed right?

If the issue has to do with _nested objects_ and how they are displayed, the reason they are displayed as they are is based on default values that should be shown -- not the type. An quick example that may not look complex at first is simply:
```
PS > $PSVersionTable

Name                           Value                                                                                                                                                                               
----                           -----                                                                                                                                                                               
PSVersion                      6.0.0-beta.9                                                                                                                                                                        
PSEdition                      Core                                                                                                                                                                                
GitCommitId                    v6.0.0-beta.9                                                                                                                                                                       
OS                             Linux 4.10.0-38-generic #42~16.04.1-Ubuntu SMP Tue Oct 10 16:32:20 UTC 2017                                                                                                         
Platform                       Unix                                                                                                                                                                                
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0...}                                                                                                                                                             
PSRemotingProtocolVersion      2.3                                                                                                                                                                                 
SerializationVersion           1.1.0.1                                                                                                                                                                             
WSManStackVersion              3.0
```
Which is just a hashtable of multiple objects that differ when it comes to object types. We see summarized _values_. `PSCompatibleVersions` is a list of `System.Version` object types, displayed in a summarized/truncated fashion. The objects themselves look entirely different in output when called directly:
```
PS > $PSVersionTable.PSCompatibleVersions                                                                                                                                                            

Major  Minor  Build  Revision
-----  -----  -----  --------
1      0      -1     -1      
2      0      -1     -1      
3      0      -1     -1      
4      0      -1     -1      
5      0      -1     -1      
5      1      10032  0       
6      0      0      -1
```

Even in this case, the console is showing a reduced list of properties:

```
PS > $PSVersionTable.PSCompatibleVersions | select -First 1                                                                                                                                          

Major  Minor  Build  Revision
-----  -----  -----  --------
1      0      -1     -1

PS > $PSVersionTable.PSCompatibleVersions | select -First 1 | Get-Member                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                        TypeName: System.Version                                                                                                                                                                                                                                                                                                                                                                                                             
Name          MemberType Definition                                                                                                                                                         
----          ---------- ----------                                                                                                                                                         
Clone         Method     System.Object Clone(), System.Object ICloneable.Clone()                                                                                                            
CompareTo     Method     int CompareTo(System.Object version), int CompareTo(version value), int IComparable.CompareTo(System.Object obj), int IComparable[version].CompareTo(version other)
Equals        Method     bool Equals(System.Object obj), bool Equals(version obj), bool IEquatable[version].Equals(version other)                                                           
GetHashCode   Method     int GetHashCode()                                                                                                                                                  
GetType       Method     type GetType()                                                                                                                                                     
ToString      Method     string ToString(), string ToString(int fieldCount)                                                                                                                 
Build         Property   int Build {get;}                                                                                                                                                   
Major         Property   int Major {get;}                                                                                                                                                   
MajorRevision Property   int16 MajorRevision {get;}                                                                                                                                         
Minor         Property   int Minor {get;}                                                                                                                                                   
MinorRevision Property   int16 MinorRevision {get;}                                                                                                                                         
Revision      Property   int Revision {get;}

PS > $PSVersionTable.PSCompatibleVersions | select -First 1 | select *                                                                                                                                                 


Major         : 1
Minor         : 0
Build         : -1
Revision      : -1
MajorRevision : -1
MinorRevision : -1
```

The solution to _"wait, there's more, I'm not just a string, int32, or double and I even have properties that don't output to console by default..."_ would be learning the value of `Get-Member` to learn more. `$PSVersionTable` is a mixed bag of stuff when it comes to types:

```
$PSVersionTable.values | get-member | select TypeName -Unique                                                                                                                                   

TypeName                                    
--------                                    
System.Management.Automation.SemanticVersion
System.String                               
System.Version[]                            
System.Version
```

There are [video courses](https://mva.microsoft.com/search/SearchResults.aspx#!q=powershell&lang=1033), [technical books](https://www.amazon.com/Learn-Windows-PowerShell-Month-Lunches), [docs/learning-powershell in this same repository](https://github.com/PowerShell/PowerShell/blob/master/docs/learning-powershell/working-with-powershell-objects.md), [interactive free on-demand virtual labs](https://vlabs.holsystems.com/vlabs/technet?eng=VLabs&auth=none&src=vlabs&altadd=true&labid=20434&lod=true), and other places that explain (how objects work in) PowerShell.

[The Monad Manifesto](http://www.jsnover.com/Docs/MonadManifesto.pdf) explains the reasoning behind why PowerShell isn't just another shell with a text manipulation pipeline. 

@markekraus If there were a hypothetical interactive tutorial for the local console, wouldn't it be better for someone to make a PowerShell Module as opposed to a modification/add to PowerShell itself as a feature? Otherwise, you just pushed a tutorial to systems where people aren't even going to need it. `Update-Help` exists to avoid carrying the help documentation everywhere by default, too, and makes it so that PowerShell itself doesn't need to update when help documentation is updated for a single module.

Should this issue be closed?

**Comment by markekraus** [comment]
> If there were a hypothetical interactive tutorial for the local console, wouldn't it be better for someone to make a PowerShell Module as opposed to a modification/add to PowerShell itself as a feature? 

That is where I would start, but, there is some merit in it being a native part of the system. The problem is that someone new to the language who just runs `pwsh` in their console for the first time probably has no clue what modules are or how to install them (and even as a veteran user, I have to say that PowerShellGallery is not the most user friendly tool once you know about it). They have several hurdles of knowledge and acclamation to jump first. If the underlying goal of @HumanEquivalentUnit's request was to make this easier for the novie user, then including an interactive tutorial with PowerShell and not apart from it makes perfect sense. 

The part about the tutorial being forced was only a jest. @iSazonov recently had #5275 merged which adds a small banner on how to get help. That banner could just be expanded slightly to include something like `or 'Start-Tuorial' for an interactive tutorial.` The interactive tutorial could very well be a separate module and only loaded when called. It could even be just a stub that begins by walking the user through installing the actual module from the the gallery, that way the initial footprint could be small.

In any case, this is all just brain storming.

> Should this issue be closed?
I have changed the label to be a discussion. I'll leave this open for a few more days to see if there is any more feed back and then close it. The consensus appears to be that `select-object` shouldn't be changed, but the discussion as started to hit at solutions for the root cause.

**Comment by ScriptAutomate** [comment]
@markekraus Okay, cool. That makes sense.

I noticed the new small banner for help, which looks great. Reminds me of ipython's initial banner:
```
Python 3.6.2 (default, Sep 10 2017, 19:48:49) 
Type 'copyright', 'credits' or 'license' for more information
IPython 6.2.1 -- An enhanced Interactive Python. Type '?' for help.

In [1]: 
```
I like the idea of it being a stub for installing the tutorial, perhaps walking the user through basic commands and diving into them as objects via the pipeline. The idea sort of reminds me of the `Start-Demo` module, if perhaps there was a default demo maybe. I haven't tested that module out in a good while, but it may be a starting point?

**Comment by SteveL-MSFT** [comment]
I like the idea of `Start-Tutorial` (perhaps a community driven module...), although not keen on having it in the core package.  Would prefer we add the feature https://github.com/PowerShell/PowerShell/issues/5523 so perhaps the first time they run `pwsh`, we can suggest they try `Start-Tutorial` if this is their first time.

**Comment by BrucePay** [comment]
We should probably note somewhere (select-object docs?) that if you just want to get the value of a single property, you should do
`ls | foreach name`
instead of
`ls | select name`

**Comment by SteveL-MSFT** [comment]
@BrucePay you should probably submit a PR to fix `Select-Object` [doc](https://github.com/PowerShell/PowerShell-Docs/blob/staging/reference/6/Microsoft.PowerShell.Utility/Select-Object.md) which is where people might see this

**Comment by Pxtl** [comment]
Could we get new built-in commandlets to make this easy?

`Select-PropertyExpanded` which would work the same as `Select-Object -ExpandProperty` except the default parameter would be `-ExpandProperty` and the `Property` member wouldn't exist?

With a nice alias of `expandprop`?

So we can just say `ls | expandprop Name`

**Comment by Website-monitoringdour** [comment]
https://practicablemsgpack.github.io/9m39w4DtJH5CzIYIoSNdwhCIMAspqCqpEEot1Q3yf33atHG6uIXwFRLS/

<blockquote></blockquote>
